File: Searching/BinarySearch.java
public class BinarySearch {

    public static int binarySearch(int arr[], int target) {
        int low=0 , high = arr.length-1;

        while(low <= high) {
            int mid = low+(high-low)/2;
            if(arr[mid] == target) {
                return mid;
            }
            else if(arr[mid] < target) {
                low = mid+1;
            }
            else {
                high = mid-1;
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        int arr[] = {10, 12, 45, 48, 56, 68, 98, 104};
        int target = 10;
         
        int result = binarySearch(arr, target);
         
        if(result == -1) {
            System.out.println("element is not found in an array");
        }
        else{
            System.out.println("Element found at "+result+" index");
        }
    }
}

File: Searching C/BinarySearch.c
#include <stdio.h>

int binarySearch(int arr[], int target, int n) {
    int low =0, high = n;

    while(low < high) {
        int mid = low+(high-low)/2;
        if(arr[mid] == target) {
            return mid;
        }
        else if(arr[mid] < target) {
            low = mid+1;
        }
        else{
            high = mid-1;
        }
    }
    return -1;
}
int main() {

    int arr[] = {5, 8, 10, 23, 47, 78, 97};
    int n = sizeof(arr)/sizeof(arr[0]);
    int target = 8;

    int result = binarySearch(arr, target, n);

    if(result == -1) {
        printf("Element is not found in an array.");
    }
    else{
        printf("element found at %d index.", result);
    }
    return 0;
}